app/controllers/api/v1/transcriptions_controller.rb

```diff
@@ -6,10 +6,17 @@ class Api::V1::TranscriptionsController < Api::BaseController
   def create
     page = Page.find_by(id: params[:page_id])
     if page
-      transcription = page.transcriptions.create!(transcription_params.merge(user: current_user))
-      text = ai_transcribe(params[:transcription][:audio_file])
-      transcription.update!(transcription_text: text, status: :transcribed)
-      render json: format_transcription(transcription), status: :created
+      begin
+        transcription = page.transcriptions.create!(transcription_params.merge(user: current_user))
+        text = ai_transcribe(params[:transcription][:audio_file])
+        transcription.update!(transcription_text: text, status: :transcribed)
+      rescue StandardError => e
+        Rails.logger.error("Error during transcription processing: #{e.message}")
+        transcription.update!(status: :failed)
+        render json: { error: "Internal server error occurred while processing transcription" }, status: :internal_server_error
+      else
```

Comment:
Maybe we extract it into a method that can be used by other classes 🤔 
```suggestion
      rescue StandardError => e
    handle_error(transcription, e, "Error during transcription processing")
      else
```
```rb
  def handle_error(transcription, error, message)
    Rails.logger.error("#{message}: #{error.message}")
    transcription.update!(status: :failed)
    render_error("Internal server error occurred while processing transcription", :internal_server_error)
  end
  
  def render_error(message, status)
    render json: { error: message }, status: status
  end
  ```

Reply:
I would suggest implementing a generic error handler class that can be reused across all controllers. This approach would not only minimize the need to handle errors individually within each controller but also significantly reduce code duplication, making the overall codebase cleaner and more maintainable.